maxim/logger/livekit/gemini/gemini_realtime_session.py

```
@@ -37,18 +45,29 @@ def handle_build_connect_config_result(
-    llm_config["speech_config"] = result.speech_config.model_dump()
+    session_info.provider = "google-realtime"
+    llm_config = {}
+    if result.generation_config is not None:
+        llm_config = result.generation_config.model_dump()
+    if result.speech_config is not None:
+        for key, value in result.speech_config.model_dump().items():
+            if value is not None:
+                llm_config[key] = value
+    # removing speech_config from the llm_config
+    llm_config.pop("speech_config", None)
     llm_config["model"] = self._opts.model
```

Pylint (3.3.7)

[warning] Access to a protected member _opts of a client class

(W0212)